### PR TITLE
fix: failing integration tests 

### PR DESCRIPTION
**Pull request by cka-y**
**Summary:**

- **Fixed Failing Integration Tests:**
  - Implemented an outer join for the `location` and `feed` entities to ensure proper data fetching.
  - Added `joinedload` for all entities accessed by the `from_orm` methods to optimize query performance and reduce additional database queries. For instance, querying `https://api-qa.mobilitydatabase.org/v1/gtfs_feeds` now takes approximately 1.6 seconds, down from 17 seconds.
- Removed the unnecessary test for the validity of the latest URL, as this validation is already handled within the data pipeline. Performing this test within the integration tests is redundant and not the most appropriate place for it.

[Example of successful execution](https://github.com/MobilityData/mobility-feed-api/actions/runs/9863354004/job/27236236214)

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: api/src/feeds/impl/feeds_api_impl.py [hunk 1]
```diff
@@ -155,6 +161,21 @@ def get_gtfs_feeds(
...
+        gtfs_feed_query = (
+            gtfs_feed_query.outerjoin(Location, Feed.locations)
+            .outerjoin(latest_datasets, Gtfsfeed.gtfsdatasets)
+            .options(
+                contains_eager(Gtfsfeed.gtfsdatasets, alias=latest_datasets)
+                .joinedload(Gtfsdataset.validation_reports)
+                .joinedload(Validationreport.notices),
+                *BasicFeedImpl.get_joinedload_options(),
+            )
+        )
```
Comment (davidgamez) [inline api/src/feeds/impl/feeds_api_impl.py:178]: How can we benefit from the latest datasets query if the _GtfsRTFeedImpl.from_orm_ is still looping over all datasets?

Comment (cka-y) [reply api/src/feeds/impl/feeds_api_impl.py]: Indeed! There's no way to load the latest dataset exclusively so I removed this query.

Comment (davidgamez) [reply api/src/feeds/impl/feeds_api_impl.py]: We will hit a performance issue in the future as the datasets keep growing. We can revisit mapping the latest dataset as part of the entity. I remember having a few challenges with the previous SqlAchemy version; maybe the new version is easier to implement. Created an issue for it #548

**Review by davidgamez — APPROVED** [review]
LGTM!

**Pull request merged**